Instructions at signup

apapadoi opened the pull request:
Instructions are now added on the signup page to help users get their grades from sis and upload them on our app.

This pr closes #40.

apapadoi committed:
Completed first version of instructions on how to get your grades from sis at signup page.

apapadoi committed:
Added documentation and refactored the code.

apapadoi commented:
Known issue : Gradient color does not change correctly when user chooses dark theme and logs out from the app.

sakrifor committed:
Minor Text Changes

sakrifor commented on lib/pages/general/SignUpInstructionsPage.dart:
```diff
@@ -0,0 +1,154 @@
+  ];
+
+  final List<String> descriptionsList = [
+    'Visit https://sis.auth.gr/old and login with your academic account.',
```
This should be a link. Maybe if it works we can change it to https://sis.auth.gr/old/el/all_grades and reduce the number of pages.

apapadoi replied:
Commit with id 0db22d now meets the proposed change.

![image](https://user-images.githubusercontent.com/57541732/118804595-8ed35680-b8ad-11eb-9a1c-f4d938b4f7a3.png)

sakrifor commented on lib/pages/general/SignUpInstructionsPage.dart:
```diff
@@ -0,0 +1,154 @@
+                else
+                  SizedBox(),
+                if (imgIndex != imgList.length - 1)
+                  _buildTextButton('Next', _nextImage)
```
Maybe we can remove the buttons and make it with swipes? And a final ok at the end.

apapadoi replied:
Commit with id 0a39764 now meets the proposed change.

https://user-images.githubusercontent.com/57541732/118834345-d3b9b600-b8ca-11eb-9435-518158e6dfa1.mp4

sakrifor commented on lib/pages/authentication/AuthPage.dart:
```diff
@@ -83,6 +83,19 @@ class _AuthPageState extends PageState<AuthPage> {
                   ],
                 ),
               ),
+              Align(
+                alignment: Alignment.bottomCenter,
+                child: TextButton(
+                  child: StyledText(
+                    'Need help uploading your grades? Find more info here!',
+                    size: 14,
+                    align: TextAlign.center,
+                    color: Colors.blue[400],
+                  ),
+                  onPressed: () =>
+                      nav.Router.push(context, '/signup_instructions'),
+                ),
+              ),
             ],
```
This should be moved inside the form next or below the upload button

apapadoi committed:
Changed the text url, in the first step of the instructions page on how to get your grades from sis, in a tappable link that opens in a brouwser tab.

apapadoi committed:
Removed buttons and added swipe left/right functionality in the instructions page and a final ok button that redirects the user to the AuthPage.

sakrifor committed:
Moving to form

sakrifor reviewed (changes requested):
I've moved the carousel inside the form. There are still minor styling problems. Please test in on small screen cause there are some overflows and use the same style as the other pages (button texts etc.) Please don't use custom colors.

apapadoi committed:
Fixed pixel overflow in the link of the instructions page.

apapadoi committed:
Removed unnecessary styling and added support for dark theme.

apapadoi commented:
Fixed pixel overflow and removed unnecessary styling. Also added support for dark theme.

![1](https://user-images.githubusercontent.com/57541732/119814403-0713ca80-bef3-11eb-924c-f8399b0fa570.png)
![2](https://user-images.githubusercontent.com/57541732/119814406-0844f780-bef3-11eb-94be-d1a68333e497.png)
![image](https://user-images.githubusercontent.com/57541732/119814421-0c711500-bef3-11eb-8346-4cd37ca5a880.png)
![image](https://user-images.githubusercontent.com/57541732/119814428-0f6c0580-bef3-11eb-994f-6e44cef5c059.png)
![image](https://user-images.githubusercontent.com/57541732/119814441-1266f600-bef3-11eb-9b19-174abe566e5c.png)
![image](https://user-images.githubusercontent.com/57541732/119814449-1561e680-bef3-11eb-8322-fdbf330402a8.png)
![image](https://user-images.githubusercontent.com/57541732/119814460-185cd700-bef3-11eb-956b-1c0dd273bbef.png)

apapadoi committed:
Removed font weight from text at instructions page.

sakrifor committed:
Fix dispose warning on animationController

sakrifor committed:
Code style fixes

sakrifor committed:
PubSpec Upgrade

sakrifor reviewed (approved):
Very nice job @TasosOperatingInBinary ! While there are some issues still it is better to accept this and make a mockup first.
